megamek/src/megamek/common/pathfinder/AeroGroundPathFinder.java

```diff
@@ -0,0 +1,647 @@
+            Entity currentEntity = entityIter.next();
+            if(currentEntity.isAero()) {
+                return true;
+            }
+        }
+        
+        return false;
+    }
+    
+    /**
+     * Helper function to calculate the maximum thrust we should use for a particular aircraft
+     * We limit ourselves to the lowest of "safe thrust" and "structural integrity"...
+     *  ... or something a little lower to prevent using gigabyte after gigabyte of memory on high-thrust paths
+     *  that don't do anything useful.
+     * @param aero The aero entity for which to calculate max thrust.
+     * @return The max thrust.
+     */
+    private static int calculateMaxThrust(IAero aero) {
+        int maxThrust = Math.min(aero.getCurrentThrust(), aero.getSI());    // we should only thrust up to our SI
+        //maxThrust = Math.min(maxThrust, artificialCap); // lol just kidding, you ain't going nowhere fast bro
+        return maxThrust;
+    }
+    
+    /**
+     * 
+     * @author NickAragua
+     *
+     * @param <MovePath> a move path
+     * 
+     * This class removes all off-board paths, but does kee
+     */
+    public static class AeroGroundOffBoardFilter extends Filter<MovePath> {
+
+        MovePath shortestPath;
+        
+        public MovePath getShortestPath() {
+            return shortestPath;
+        }
+        
+        /**
+         * Returns filtered collection by removing those objects that fail
+         * {@link #shouldStay(T)} test.
+         * 
+         * @param collection collection to be filtered
+         * @return filtered collection
+         */
+        @Override
+        public Collection<MovePath> doFilter(Collection<MovePath> collection) {
+            List<MovePath> filteredMoves = new ArrayList<>();
+            for (MovePath e : collection) {
+                // if the path does not go off board, then we add it to the returned collection
+                if (shouldStay(e)) {
+                    filteredMoves.add(e);
+                }
+                else {
+                    // if the path *does* go off board, we want to compare its length to 
+                    // our current shortest off-board path and keep it in mind if it's shorter
+                    // at the end of the process, we will add the shortest path off board to the list of moves
+                    if(shortestPath == null || 
+                            (shortestPath.length() > e.length())) {
+                        shortestPath = e;
+                    }
+                }
+            }
+            
+            return filteredMoves;
+        }
+        
+        /**
+         * Tests if the object should stay in the collection.
+         * 
+         * @param object tested object
+         * @return true if the object should stay in the collection
+         */
+        @Override
+        public boolean shouldStay(MovePath object) {
+            MovePath path = (MovePath) object;
```

Comment:
What's the purpose of this?

Reply:
The basic idea is that, practically speaking, all off-board paths are the same, so we eliminate all but the shortest one from consideration.

Reply:
I was specifically referring to the creation of the second path reference.  Near as I can tell, line 189 doesn't really do anything useful.

Reply:
Ah, I see. It's probably a holdover from when that object wasn't explicitly a MovePath when I first wrote it. I'll get rid of it.